avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroGenerator.java

```diff
@@ -382,6 +390,17 @@ public final void writeStartObject() throws IOException {
     }
 
     @Override
+    public void writeStartObject(Object forValue) throws IOException {
```

Comment:
Ah good idea. While method was originally added for slightly different purpose (simply to allow keeping track of "current POJO"), it makes sense, and is great additional use.
The only (?) potential concern is to ensure that this method gets always called; mostly concerned wrt traversal through array/Collection serializers or such.